### ISSUE TITLE
TargetFramework net7.0 disables all CAxxxx rules

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn-analyzers#6245]
Just bumped my project to `.NET 7.0 RC2` (running `VS Version 17.4.0 Preview 6.0`), and realized that all code analysis rules are disabled, no matter the settings. Downgrading the project to `net6.0` re-enables all CAxxxx rules.

### Steps To Reproduce
Just create a console project with some CAxxxx violations, and target `net7.0`. No warnings are produced. Downgrade to `net6.0` to get the warnings back.

```xml
<PropertyGroup>
   <!--TargetFramework>net6.0</TargetFramework-->
   <TargetFramework>net7.0</TargetFramework>

   <OutputType>Exe</OutputType>
   <LangVersion>latest</LangVersion>
   <ImplicitUsings>enable</ImplicitUsings>
   <Nullable>enable</Nullable>
   <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
   <AnalysisMode>AllEnabledByDefault</AnalysisMode>
   <WarningLevel>9999</WarningLevel>
</PropertyGroup>
```

### Expected behavior
CAxxxx rules are checked when targeting `.NET 7`.

### Actual behavior
CAxxxx rules are checked only when targeting `.NET 6` or earlier.

**Comment by mavasani** [comment]
@znakeeye Thank you so much for reporting the issue. I have created a PR that should fix this issue once the fix flows into dotnet/sdk repo. @jmarolf Can we ensure the flow into the SDK repo happens asap? Do we need any approvals here?

As a side note, I am going to enhance the tooling so such a regression does not happen in future. Our generated editorconfig files that we ship in the package should most likely include a `vNext` set of editorconfig files so that when the analysis level is set to `latest` or explicitly to `vNext`, we still find a proper mapped editorconfig file.

**Comment by mavasani** [comment]
Filed #6247

**Comment by znakeeye** [comment]
Many thanks for the fix. I actually saw this problem some months back, but went back to `net6.0` and forgot about it. Glad I reported it 😄
